### PR TITLE
LANG-701 Added new joinWith method to support varargs

### PR DESCRIPTION
**Pull request by jamessawle**
...rom previous join methods, but have added ExpectedException rule to allow for better exception handling in tests

### CODE REVIEW

**Comment by britter** [comment]
Regarding coveralls: see my comment on #63 :)

File: src/test/java/org/apache/commons/lang3/StringUtilsTest.java [hunk 1]
```diff
@@ -52,32 +52,6 @@
...
-        for (int i = 0; i <= 32; i++) {
-            tr += String.valueOf((char) i);
-        }
-        WHITESPACE = ws;
-        NON_WHITESPACE = nws;
-        HARD_SPACE = hs;
-        TRIMMABLE = tr;
-        NON_TRIMMABLE = ntr;
-    }
-
```
Comment (britter) [inline src/test/java/org/apache/commons/lang3/StringUtilsTest.java]: Please don't rearrange code if it is not necessary. It makes reviewing patches harder.

File: src/test/java/org/apache/commons/lang3/StringUtilsTest.java [hunk 2]
```diff
@@ -460,6 +456,24 @@ public void testJoin_IterableString() {
...
+
+		assertEquals("ab", StringUtils.joinWith(null, "a", "b"));
+	}
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testJoinWithThrowsException() {
+        StringUtils.joinWith(",", null);
+	}
+
+
```
Comment (britter) [inline src/test/java/org/apache/commons/lang3/StringUtilsTest.java]: No tabs please.

**Comment by jamessawle** [comment]
Sorry, I will fix this this afternoon. I forgot to turn off my companies coding standards before I committed and this did the re-arrangements

**Comment by jamessawle** [comment]
Changes have been made. Sorry forgot to reset my auto-code format settings

**Comment by britter** [comment]
@jamessawle Commons Lang has been migrated from svn to git. Because of this the main branch has been renamed to 'master'. Can you please rebase your work against the master branch? Thank you!

**Comment by jamessawle** [comment]
Created new pull request

**Pull request closed without merging**